### PR TITLE
Add tools to ingest data, produced by an NSAPH containerized app into Dorieh data warehouse

### PR DESCRIPTION
**Pull request by mmcentre**
See https://github.com/NSAPH-Data-Platform/nsaph-core-platform/blob/app_ingestion_1/doc/AppPipelineGenerator.md

### CODE REVIEW

**Comment by mmcentre** [comment]
Related PR: https://github.com/NSAPH-Data-Processing/zip2zcta_master_xwalk/pull/15

File: doc/AppPipelineGenerator.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+using `master` branch and output the result into current directory in
+a file named `pipeline.cwl`
+
+Alternatively, to do the same using Docker container, execute:
+
+    docker run -v $(pwd):/tmp/work forome/dorieh python -m nsaph.apprunner.app_run_generator https://github.com/NSAPH-Data-Processing/zip2zcta_master_xwalk.git /tmp/work/pipeline.cwl master
+
+In both cases, the generator will produce 3 files:
+
+* pipeline.cwl: main workflow file
```
Comment (audiracmichelle) [inline doc/AppPipelineGenerator.md:96]: interesting solution :)

File: doc/AppPipelineGenerator.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+ 
+## Execute generated pipeline
+
+If you installed Dorieh packages in your local Python virtual environment, 
+you can execute the pipeline with the following command in 
+the working directory, for example using CWL reference implementation
+built into Dorieh (cwl-runner).
+
+    cwl-runner pipeline.cwl --database $path_to_your connection_def_file --connection_name $connection_name
+
```
Comment (audiracmichelle) [inline doc/AppPipelineGenerator.md:111]: small typo: $path_to_dbini instead of $path_to_your

File: doc/AppPipelineGenerator.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+A better way would be to use a production grade CWL implementation
+such as [Toil](https://toil.readthedocs.io/en/latest/running/cwl.html).
+To do this you need to 
+[install Toil](https://toil.readthedocs.io/en/latest/gettingStarted/install.html) 
+on your local system. 
+
+> You do not need to install Dorieh packages to execute the pipeline.
+> The runtime engine will use Dorieh container where all requirements
+> are preinstalled.
+
```
Comment (audiracmichelle) [inline doc/AppPipelineGenerator.md:125]: I am behind on this, I am not sure where is the Dorieh container being generated.

File: setup.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 setup(
     name='nsaph',
-    version="0.0.4.6",
+    version="0.0.4.7",
```
Comment (audiracmichelle) [inline setup.py]: I am familiar with github releases for stamped releases, is this an alternative?

File: setup.py [hunk 5]
```diff
@@ -18,15 +18,15 @@
...
+        'nsaph.util', 'nsaph.dbt', 'nsaph.apprunner',
+        'nsaph_resources', 'nsaph_models', 'nsaph.sql', 'nsaph.commonwl'
     ],
     package_dir={
         "nsaph": "./src/python/nsaph",
         "nsaph_models": "./src/yml",
         "nsaph.sql": "./src/sql",
         "nsaph_resources": "./resources",
-        "nsaph_tools": "./src/cwl"
+        "nsaph.commonwl": "./src/commonwl"
```
Comment (audiracmichelle) [inline setup.py:30]: I had not seen this, are this the paths in nsaph node?

File: src/commonwl/ingest.cwl [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+class: CommandLineTool
+baseCommand: [python, -m, nsaph.loader.data_loader]
+# baseCommand: echo
+requirements:
+  InlineJavascriptRequirement: {}
+
+hints:
+  DockerRequirement:
+    dockerPull: forome/dorieh
+
```
Comment (audiracmichelle) [inline src/commonwl/ingest.cwl:32]: should I be pulling this container for all dorieh cwl runs?

**Comment by audiracmichelle** [comment]
My understanding is that the tool writes a cwl file with dataset specific content which can then be passed to another cwl.

Am I understanding correctly?

**Comment by audiracmichelle** [comment]
@mmcentre I think I have to pull all packages to resolve the conflict, correct?
```
    packages=[
        'nsaph', 'nsaph.requests', 'nsaph.data_model', 'nsaph.loader',
<<<<<<< app_ingestion_1
        'nsaph.util', 'nsaph.dbt', 'nsaph.apprunner',
        'nsaph_resources', 'nsaph_models', 'nsaph.sql', 'nsaph.commonwl'
=======
        'nsaph.util', 'nsaph.dbt',
        'nsaph.operations',
        'nsaph_resources', 'nsaph_models', 'nsaph.sql'
>>>>>>> develop
    ],
    ```

**Comment by audiracmichelle** [comment]
w

> @mmcentre I think I have to pull all packages to resolve the conflict, correct?
> 
> ```
>     packages=[
>         'nsaph', 'nsaph.requests', 'nsaph.data_model', 'nsaph.loader',
> <<<<<<< app_ingestion_1
>         'nsaph.util', 'nsaph.dbt', 'nsaph.apprunner',
>         'nsaph_resources', 'nsaph_models', 'nsaph.sql', 'nsaph.commonwl'
> =======
>         'nsaph.util', 'nsaph.dbt',
>         'nsaph.operations',
>         'nsaph_resources', 'nsaph_models', 'nsaph.sql'
> >>>>>>> develop
>     ],
>     ```
> ```

Would be good to understand what nsaph.operations stand for too

**Pull request merged**